addons/payment/controllers/portal.py

```diff
@@ -222,7 +222,14 @@ def pay(self, reference='', order_id=None, amount=False, currency_id=None, acqui
         # s2s mode will always generate a token, which we don't want for public users
         valid_flows = ['form', 's2s'] if not user._is_public() else ['form']
         values['acquirers'] = [acq for acq in acquirers if acq.payment_flow in valid_flows]
-        values['pms'] = request.env['payment.token'].search([('acquirer_id', 'in', acquirers.ids)])
+        if partner_id:
+            values['pms'] = request.env['payment.token'].search([
+                ('acquirer_id', 'in', acquirers.ids),
+                ('partner_id', '=', partner_id)
+            ])
+        else:
+            values['pms'] = False
```

Comment:
i'd rather it was an empty iterable, i'm not sure it never gets iterated upon
```suggestion
            values['pms'] = []
```

Reply:
ideally perhaps an empy recordset of the token model

Reply:
It's iterate on the view payment.payment_tokens_list but False doesn't seems an issue for qweb but I agree an empty iterable would be better

Reply:
lol i don't think i needed to be included as a co-author